### DISCUSSION TITLE
BorrowError at Compile for Paths?

### DISCUSSION THREAD

**Discussion by Farmbot26** [slint-ui/slint#2462] [Q&A]
I'm very confused. I have a ModelRc of structs that gets populated from an HTTP request. Each struct is meant to get a Path drawn in the UI.

This runs correctly:

for field[i] in select-map(map-combo.current-index).fields : Text {
       text: field.commands;
}

This crashes on compile:

for field[i] in select-map(map-combo.current-index).fields : Path {
       commands: field.commands;
       stroke: green;
       stroke-width: 1px;
}

with this error: thread 'main' panicked at 'already mutably borrowed: BorrowError'

Can anyone help?

**Comment by tronical** [comment]
That sounds like a bug. Could you please post a backtrace? (Maybe setting RUST_BACKTRACE=1 will help showing it)

**Reply by Farmbot26** [reply]
Full Stack Trace (with my name removed):  

thread 'main' panicked at 'already mutably borrowed: BorrowError', /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/expression_tree.rs:571:22
  stack backtrace:
     0:     0x55c2336da9da - std::backtrace_rs::backtrace::libunwind::trace::h8f4e76b8218b8540
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
     1:     0x55c2336da9da - std::backtrace_rs::backtrace::trace_unsynchronized::hcdb6a06d2fd94ab3
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
     2:     0x55c2336da9da - std::sys_common::backtrace::_print_fmt::h6986e85b14c0a3c2
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:65:5
     3:     0x55c2336da9da - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hf09049473c7eb0d5
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:44:22
     4:     0x55c2336fd6ce - core::fmt::write::h7b08166a7a14a28e
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/fmt/mod.rs:1208:17
     5:     0x55c2336d7755 - std::io::Write::write_fmt::hdb3e773b5b1c54df
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/io/mod.rs:1682:15
     6:     0x55c2336da7a5 - std::sys_common::backtrace::_print::h1ceaebbb26a3fab3
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:47:5
     7:     0x55c2336da7a5 - std::sys_common::backtrace::print::hb5aa1269b1b9bc74
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:34:9
     8:     0x55c2336dbf8f - std::panicking::default_hook::{{closure}}::hbaa8ce6d79af00fb
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:267:22
     9:     0x55c2336dbccb - std::panicking::default_hook::hcd65d80b40db7fc0
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:286:9
    10:     0x55c2336dc69c - std::panicking::rust_panic_with_hook::hbb6ac4823ece4d03
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:688:13
    11:     0x55c2336dc439 - std::panicking::begin_panic_handler::{{closure}}::h325f7839ba4584c6
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:579:13
    12:     0x55c2336dae8c - std::sys_common::backtrace::__rust_end_short_backtrace::hacaef3341973bb56
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:137:18
    13:     0x55c2336dc142 - rust_begin_unwind
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:575:5
    14:     0x55c2328006a3 - core::panicking::panic_fmt::hef6845e1540bf16b
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/panicking.rs:64:14
    15:     0x55c232800b33 - core::result::unwrap_failed::h1f98fd83f4600fae
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/result.rs:1791:5
    16:     0x55c232d344be - core::result::Result<T,E>::expect::hd6d6e72c686689d0
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/result.rs:1070:23
    17:     0x55c232bc9fc7 - core::cell::RefCell<T>::borrow::h45ecbc94fb584d22
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/cell.rs:865:9
    18:     0x55c232a8f2b0 - i_slint_compiler::expression_tree::Expression::ty::h34c23ecbbb5c2c7f
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/expression_tree.rs:568:56
    19:     0x55c232a8dc5b - i_slint_compiler::expression_tree::Expression::ty::h34c23ecbbb5c2c7f
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/expression_tree.rs:586:67
    20:     0x55c232ca92bf - i_slint_compiler::passes::compile_paths::compile_paths::{{closure}}::h403ffc01bb630f95
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/passes/compile_paths.rs:70:25
    21:     0x55c232a287ff - i_slint_compiler::object_tree::recurse_elem::hcb1edd9ce94ac36a
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/object_tree.rs:1791:17
    22:     0x55c232a28977 - i_slint_compiler::object_tree::recurse_elem::hcb1edd9ce94ac36a
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/object_tree.rs:1793:9
    23:     0x55c232a28977 - i_slint_compiler::object_tree::recurse_elem::hcb1edd9ce94ac36a
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/object_tree.rs:1793:9
    24:     0x55c232a28977 - i_slint_compiler::object_tree::recurse_elem::hcb1edd9ce94ac36a
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/object_tree.rs:1793:9
    25:     0x55c232ca7f6d - i_slint_compiler::passes::compile_paths::compile_paths::h133739e0137095b2
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/passes/compile_paths.rs:28:5
    26:     0x55c232819368 - i_slint_compiler::passes::run_passes::{{closure}}::h34fc1f852cd13790
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/passes.rs:85:9
    27:     0x55c23283c117 - i_slint_compiler::compile_syntax_node::{{closure}}::hf1870a4d78356b8a
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/i-slint-compiler-0.3.5/lib.rs:181:82
    28:     0x55c232838300 - spin_on::spin_on::hac5e530c3ca98287
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/spin_on-0.1.1/src/lib.rs:78:38
    29:     0x55c232838e6b - slint_build::compile_with_config::heecd61188d18ecb0
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/slint-build-0.3.5/lib.rs:323:9
    30:     0x55c23283a8bd - slint_build::compile::h6f6839dc748cb4db
                                 at /home/name/.cargo/registry/src/github.com-1ecc6299db9ec823/slint-build-0.3.5/lib.rs:272:5
    31:     0x55c232847ada - build_script_build::main::hb521439a8378dd4d
                                 at /home/name/Desktop/drip_truck/build.rs:2:5
    32:     0x55c23280732b - core::ops::function::FnOnce::call_once::hbaf360911fa3c70d
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/ops/function.rs:507:5
    33:     0x55c232804e9e - std::sys_common::backtrace::__rust_begin_short_backtrace::hfa9a873f9bdd51b7
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/sys_common/backtrace.rs:121:18
    34:     0x55c23283cc61 - std::rt::lang_start::{{closure}}::he917a268fa2aa351
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/rt.rs:166:18
    35:     0x55c2336d246c - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h2e730c3fa855d1cd
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/core/src/ops/function.rs:606:13
    36:     0x55c2336d246c - std::panicking::try::do_call::h2452817705661d69
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:483:40
    37:     0x55c2336d246c - std::panicking::try::had9be61d7261a8f5
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:447:19
    38:     0x55c2336d246c - std::panic::catch_unwind::h0e0ee12d5e0e8c35
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panic.rs:137:14
    39:     0x55c2336d246c - std::rt::lang_start_internal::{{closure}}::hdfa1f1493a65dae1
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/rt.rs:148:48
    40:     0x55c2336d246c - std::panicking::try::do_call::h5a2ac6628cbce239
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:483:40
    41:     0x55c2336d246c - std::panicking::try::h6b550c44d18ea4fc
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panicking.rs:447:19
    42:     0x55c2336d246c - std::panic::catch_unwind::hd574fc8a4fcd921a
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/panic.rs:137:14
    43:     0x55c2336d246c - std::rt::lang_start_internal::h4189ab0091a05404
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/rt.rs:148:20
    44:     0x55c23283cc3a - std::rt::lang_start::h0d49ca1dabfa098c
                                 at /rustc/d5a82bbd26e1ad8b7401f6a718a9c57c96905483/library/std/src/rt.rs:165:17
    45:     0x55c232847b0e - main
    46:     0x7f19cfe20790 - <unknown>
    47:     0x7f19cfe2084a - __libc_start_main
    48:     0x55c232800d15 - _start
    49:                0x0 - <unknown>

**Reply by tronical** [reply]
Thanks! I can reproduce that.

**Reply by tronical** [reply]
I created https://github.com/slint-ui/slint/issues/2466 and will try to track a fix there.
